ceno_zkvm/benches/fibonacci.rs

```diff
@@ -43,6 +44,36 @@ fn setup() -> (Program, Platform) {
 fn fibonacci_prove(c: &mut Criterion) {
     let (program, platform) = setup();
     for max_steps in [1usize << 20, 1usize << 21, 1usize << 22] {
+        // estimate proof size data first
+        let (sanity_check_state, _) = run_e2e_with_checkpoint::<E, Pcs>(
+            program.clone(),
+            platform.clone(),
+            vec![],
+            max_steps,
+            Checkpoint::PrepSanityCheck,
+        );
+        assert!(
```

Comment:
You can combine your `assert` and your `if let`, since they are checking for the same condition.

Something like `let (proof, verifier)) = sanity_check_state.expect("PrepSanityCheck does not provide proof and verifier");` should do the trick.  (But I don't care about the specifics.  It's just weird to have the same check twice.)